cmds/core/tee/tee_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func TestTee(t *testing.T) {
+	tempDir := t.TempDir()
+
+	tests := []struct {
+		input  string
+		args   []string
+		append bool
+	}{
+		{
+			"hello",
+			[]string{filepath.Join(tempDir, "a1"), filepath.Join(tempDir, "a2")},
+			false,
+		},
+		{
+			"a\nb\n",
+			[]string{filepath.Join(tempDir, "b1")},
+			true,
+		},
+	}
+
+	for _, test := range tests {
+		var stdout bytes.Buffer
+		var stderr bytes.Buffer
+		cmd := New(test.append, false, test.args)
+		cmd.stdin = strings.NewReader(test.input)
+		cmd.stdout = &stdout
+		cmd.stderr = &stderr
+		err := cmd.run()
+		if err != nil {
+			t.Error(err)
+		}
+
+		for _, name := range test.args {
+			b, err := os.ReadFile(name)
+			if err != nil {
+				t.Error(err)
```

Comment:
I think that `t.Error` should be `t.Fatal` because if `err` is not nil, then the contents of `b` can't be relied on so lines 52+ aren't reliable or providing any value (and may in fact be confusing and producing the wrong error/lock of error).

Reply:
Are we care about content of b if test already failed? Probably we t.Fatal early if t.TempDir will fail.

Also I was not sure if creating t.TempDir() works well with t.Fatal, looks like it is! So t.Fatal now will clean resources, but calling os.Exit(1) will not. But previously it was a problems with cleaning resources. So I tried to avoid it.

Reply:
t.TempDir() does work nicely with tests, but in this case you have a single test case (`func TestTee(t *testing.T)`)  so the files created in each test case will *also* be visible to other test cases (which may be bad).  I think you can structure the test code to use subtests (https://go.dev/blog/subtests) to avoid this issue.

```
func TestTee(t *testing.T) {
  tests := []struct {
    input  string
    args   []string
    append bool
    }{
        {
            desc: "some descriptive name here",
            input: "hello",
            args: []{"a1","a2"},
            append: false,
            },
            // more test cases
        }
    for _, test := range tests {
      t.Run(test.desc, func(t *testing.T) {
        t := tempDir := t.TempDir() // Now tempDir will be cleaned between each test case.
      })
    }
....
```

Reply:
Made something like this, probably t.TempDir is enough.